src/v/model/fundamental.h

```diff
@@ -235,15 +235,6 @@ operator-(model::offset r, kafka::offset k) {
     return model::offset_delta{r() - k()};
 }
 
-/// \brief offset boundary type
-///
-/// indicate whether or not the offset that encodes the end of the offset
-/// range belongs to the offset range
-enum class boundary_type {
-    exclusive,
-    inclusive,
-};
```

Comment:
Yea, it isn't quite the same, but it certainly feels like we could avoid this with a more extensive interval/range type e.g. https://github.com/redpanda-data/redpanda/blob/dev/src/v/model/offset_interval.h (but maybe in utils, so we can use kafka::offset too)